CMakeLists.txt

```diff
@@ -231,325 +234,330 @@ if(FF_USE_NCCL)
     -DFF_USE_NCCL)
 endif()
 
-# Legion
-include(legion)
-
-# json
-include(json)
+# Build Legion only, not build FlexFlow
+if(BUILD_LEGION_ONLY)
+  include(legion)
```

Comment:
to avoid code repetition, you can just pull the `include(legion)` outside of the if statement, and have the rest of the code within a single if statement checking `if(NOT BUILD_LEGION_ONLY)` with no else clause